schemas/com.snowplowanalytics.snowplow/browser_context/jsonschema/1-0-0

```diff
@@ -0,0 +1,76 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for browser contexts",
+  "self": {
+    "vendor": "com.snowplowanalytics.snowplow",
+    "name": "browser_context",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+  "type": "object",
+  "properties": {
+    "viewport": {
+      "type": "string",
+      "maxLength": 20,
+      "description": "Viewport dimensions of the browser. Arrives in the form of WidthxHeight e.g. 1200x900."
+    },
+    "documentSize": {
+      "type": "string",
+      "maxLength": 20,
+      "description": "Document dimensions. Arrives in the form of WidthxHeight e.g. 1200x900"
+    },
+    "resolution": {
+      "type": "string",
+      "maxLength": 20,
+      "description": "Device native resolution. Arrives in the form of WidthxHeight e.g. 1200x900"
+    },
+    "colorDepth": {
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 1000,
+      "description": "The number of bits allocated to colors for a pixel in the output device, excluding the alpha channel."
+    },
+    "devicePixelRatio": {
+      "type": ["integer", "null"],
```

Comment:
Could this be a non-integer value as well? In addition to the 1 and 2 values, the docs say that "Other values may be returned as well in the case of unusually low resolution displays or, more often, when a screen has a higher pixel depth than double the standard resolution of 96 or 76 DPI.". So theoretically it could also be 0.XXX, right?

Reply:
It could be theoretically since it is the result of a division, but I am not aware of a case where CSS pixels could/would be set as 'fractional' to the physical pixels. The classic _'There is no half pixel'_, but certainly a browser engineer could have not thought about the case as well.

- Do you think we should set this as a `number` value ? 
- For special cases which are out of scope of 'normal' values, I believe it would be ok to have the validation provide a hint on this. Meaning events going to the bad rows. WDYT ?

Reply:
I don't know but the fact that the documentation for the `devicePixelRatio` mentions the weird cases like that makes me think that we should make it a `number` type. I guess I would opt for being more lenient here, there are all kinds of weird scaling setups which I don't know much about.

Reply:
Also the [docs say](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio) that it is "A double-precision floating-point value".